src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -10,18 +10,41 @@
 using System.Runtime.Serialization;
 using Microsoft.AspNet.OData.Builder;
 using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Extensions;
 using Microsoft.OData;
 using Microsoft.OData.Edm;
+using System.Linq;
 
 namespace Microsoft.AspNet.OData.Formatter.Deserialization
 {
     internal static class DeserializationHelpers
     {
         internal static void ApplyProperty(ODataProperty property, IEdmStructuredTypeReference resourceType, object resource,
             ODataDeserializerProvider deserializerProvider, ODataDeserializerContext readContext)
-        { 
-            IEdmProperty edmProperty = resourceType.FindProperty(property.Name);
-                       
+        {
+            IEdmProperty edmProperty;
+
+            // Extract edmProperty using a Case-Sensitive match
+            edmProperty = resourceType.FindProperty(property.Name);
+
+            if (readContext != null && readContext.EnableCaseInsensitiveModelBinding && edmProperty == null)
+            {
+                // if edmProperty is null, we try a Case-Insensitive match
+                var sProperties = resourceType.StructuralProperties().Where(k => k.Name.Equals(property.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
>var [](start = 16, length = 3)

We generally try to avoid the use of "var" as it makes it harder to understand what the code is expecting.

Reply:
#Resolved